### ISSUE TITLE
Azure-Identity ModuleNotFoundError: No module named 'distutils' in dependency msal-extensions (Python3.12)

### ISSUE THREAD

**Issue by mickare** [Azure/azure-sdk-for-python#32875]
- **Package Name**: [azure-identity](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/identity/azure-identity)
- **Package Version**: 1.15.1
- **Operating System**: MacOS (Darwin)
- **Python Version**: 3.12

**Describe the bug**
```python
>>> from azure.identity import DefaultAzureCredential
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "azurebug/venv/lib/python3.12/site-packages/azure/identity/__init__.py", line 10, in <module>
    from ._credentials import (
  File "azurebug/venv/lib/python3.12/site-packages/azure/identity/_credentials/__init__.py", line 11, in <module>
    from .default import DefaultAzureCredential
  File "azurebug/venv/lib/python3.12/site-packages/azure/identity/_credentials/default.py", line 20, in <module>
    from .vscode import VisualStudioCodeCredential
  File "azurebug/venv/lib/python3.12/site-packages/azure/identity/_credentials/vscode.py", line 22, in <module>
    from .._internal.macos_vscode_adapter import get_refresh_token, get_user_settings
  File "azurebug/venv/lib/python3.12/site-packages/azure/identity/_internal/macos_vscode_adapter.py", line 8, in <module>
    from msal_extensions.osx import Keychain, KeychainError
  File "azurebug/venv/lib/python3.12/site-packages/msal_extensions/__init__.py", line 11, in <module>
    from .cache_lock import CrossPlatLock
  File "azurebug/venv/lib/python3.12/site-packages/msal_extensions/cache_lock.py", line 7, in <module>
    from distutils.version import LooseVersion
ModuleNotFoundError: No module named 'distutils'
```

Importing azure-identity fails with an import error of the package `distutils`.
The package `distutils` was removed in Python3.12.

The dependency `msal-extensions` in the `azure-identity` package ([here](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/setup.py#L66)) is still using the `distutils` import.
It seems that the package [`msal-extensions`](https://github.com/AzureAD/microsoft-authentication-extensions-for-python) is unmaintained!

The issue is well known [since Nov2, 2022](https://github.com/AzureAD/microsoft-authentication-extensions-for-python/issues/118) and a [contribution](https://github.com/AzureAD/microsoft-authentication-extensions-for-python/issues/121) have been made to fix the issue (see the [open PR](https://github.com/AzureAD/microsoft-authentication-extensions-for-python/pull/120)).

**To Reproduce**
1. `python3.12 -m venv venv`
2. `source venv/bin/activate`
3. `python -m pip install azure-identity`
4. `python`
5. ```python
   >>> from azure.identity import DefaultAzureCredential
   ```
**Expected behavior**
Python3.12 should be supported because `azure-identity` has the classifier [`"Programming Language :: Python :: 3.12",`](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/setup.py#L50C9-L50C50).

This kind of error should have been covered by the unittests and CI pipeline.

**Screenshots**
None

**Additional context**
None

**Comment by edgarrmondragon** [comment]
One workaround (and maybe the reason why CI is passing) is to inject `setuptools` in the virtual environment:

```console
$ ./venv/bin/python -c 'from azure.identity import DefaultAzureCredential'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/project/venv/lib/python3.12/site-packages/azure/identity/__init__.py", line 10, in <module>
    from ._credentials import (
  File "/project/venv/lib/python3.12/site-packages/azure/identity/_credentials/__init__.py", line 11, in <module>
    from .default import DefaultAzureCredential
  File "/project/venv/lib/python3.12/site-packages/azure/identity/_credentials/default.py", line 20, in <module>
    from .vscode import VisualStudioCodeCredential
  File "/project/venv/lib/python3.12/site-packages/azure/identity/_credentials/vscode.py", line 22, in <module>
    from .._internal.macos_vscode_adapter import get_refresh_token, get_user_settings
  File "/project/venv/lib/python3.12/site-packages/azure/identity/_internal/macos_vscode_adapter.py", line 8, in <module>
    from msal_extensions.osx import Keychain, KeychainError
  File "/project/venv/lib/python3.12/site-packages/msal_extensions/__init__.py", line 11, in <module>
    from .cache_lock import CrossPlatLock
  File "/project/venv/lib/python3.12/site-packages/msal_extensions/cache_lock.py", line 7, in <module>
    from distutils.version import LooseVersion
ModuleNotFoundError: No module named 'distutils'

$ ./venv/bin/pip install setuptools

$ ./venv/bin/python -c 'from azure.identity import DefaultAzureCredential'
```

**Comment by kashifkhan** [comment]
Thank you for your feedback @mickare . We will investigate and get back to you asap.

**Comment by xiangyan99** [comment]
@rayluo could you help to take a look?

**Comment by xiangyan99** [comment]
Because this is a dependency of msal_extensions, please open an issue in https://github.com/AzureAD/microsoft-authentication-extensions-for-python repo.

**Comment by rayluo** [comment]
Fixed in [`msal_extensions` 1.1](https://pypi.org/project/msal-extensions/1.1.0/)
